fixed bug in communicationHandler and setParameters of SPI class

viveksb007 opened the pull request:
Fixes issue #243 #242 

Now single byte can be read successfully.
Please test with this [apk](https://drive.google.com/open?id=0B30IF3fyX4Q3MkhuWm9zX2stS1U).

jithinbp commented:
Hi, 
there are some issues with the scope. since it's a WIP, i'm not sure which ones are issues
* One channel should be selected on startup. 'CH1' checkbox can be enabled by default
* None of the other channels are active yet, right?
* please check the timebase slider. only the 875uS value shows a sine wave. when I roughly double it to 2mS , the plot becomes a nearly straight line (I think the data scaling may be incorrect).
* When device is disconnect while the scope is running, it should return to the main activity. It returns to the main activity when i plugged it back in
have you tried checking the time taken to read the whole buffer for this sample size?

CloudyPadmal commented on app/src/main/java/org/fossasia/pslab/activity/OscilloscopeActivity.java:
```diff
@@ -426,15 +426,19 @@ protected Void doInBackground(String... params) {
                 //no. of samples and timegap still need to be determined
                 scienceLab.captureTraces(1, 1000, 10, analogInput, false, null);
+                try {
+                    Thread.sleep((long) (1000 * 1000 * 10 * 1e-6));
```
This is `10` isn't it? :thinking:

viveksb007 replied:
Yes. Its just for testing to be clear of (samples * timeGap * 1e-3). Later it would be replaced with variables.

CloudyPadmal replied:
That's cool! :+1:

akarshan96 commented:
@jithinbp currently timebase slider is not working. Will create an issue for it once over with #240. I will work on your suggestion to enable CH1 checkbox by default and App returns back to Main Activity when the device is disconnected.
Thanks

viveksb007 commented:
@jithinbp This issue isn't related to scope. 
Still, I added a delay between captureTraces & fetch trace. And fixed communicationHandler Issue.
@akarshan96 Please open issues related to adding functionality to scope.
